pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -106,76 +121,114 @@ func (c *StaticResourceController) WithIgnoreNotFoundOnCreate() *StaticResourceC
 	return c
 }
 
+// WithConditionalResources adds a set of manifests to be created when the shouldCreateFnArg is true and should be
+// deleted when the shouldDeleteFnArg is true.
+// If shouldCreateFnArg is nil, then it is always create.
+// If shouldDeleteFnArg is nil, then it is !shouldCreateFnArg
+//  1. shouldCreateFnArg == true && shouldDeleteFnArg == true - produces an error
+//  2. shouldCreateFnArg == false && shouldDeleteFnArg == false - does nothing as expected
+//  3. shouldCreateFnArg == true applies the manifests
+//  4. shouldDeleteFnArg == true deletes the manifests
+func (c *StaticResourceController) WithConditionalResources(manifests resourceapply.AssetFunc, files []string, shouldCreateFnArg, shouldDeleteFnArg ConditionalFunction) *StaticResourceController {
```

Comment:
Worth unit testing this for nil create/delete condition functions they get set to default always true/false.

Possibly reuse https://github.com/openshift/library-go/pull/1110/files#diff-2ba2b98cfb79db1510fd030672e638dac32c02bce52f8238b5c10651fb7ded8d ?